Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -2626,6 +2630,36 @@ bool CSSPropertyParser::consumeWhiteSpaceShorthand(bool important)
     return true;
 }
 
+bool CSSPropertyParser::consumeScrollTimelineShorthand(bool important)
+{
+    CSSValueListBuilder namesList;
+    CSSValueListBuilder axesList;
+
+    do {
+        // A valid scroll-timeline-name is required.
+        if (auto name = CSSPropertyParsing::consumeSingleScrollTimelineName(m_range))
+            namesList.append(name.releaseNonNull());
+        else
+            return false;
+
+        // A scroll-timeline-axis is optional.
+        if (m_range.peek().type() != IdentToken)
+            axesList.append(CSSPrimitiveValue::create(CSSValueID::CSSValueBlock));
+        else if (auto axis = CSSPropertyParsing::consumeAxis(m_range))
+            axesList.append(axis.releaseNonNull());
+        else
+            return false;
+    } while (consumeCommaIncludingWhitespace(m_range));
+
+    if (namesList.isEmpty())
+        return false;
```

Comment:
Do we also need to filter out any junk? e.g. is `scroll-timeline: none junk` rejected? (we should have `test_invalid_value` tests for that.

If we do need to filter out junk, then `if (namesList.isEmpty() || !m_range.atEnd())` is probably the condition you want.

Reply:
I believe this is tested, there's a test for `--abc --abc` which is correctly rejected.